_posts/2018-05-24-Spring-IO-2018.md

```diff
@@ -0,0 +1,777 @@
+Especially if you keep your events for a long time, which is possible with Kafka, you might need to think about transforming existing events.
+
+But these messaging systems tend to be designed for scale.
+
+Pro tip: make use of [JsonPath](https://github.com/json-path/JsonPath){:target="_blank"} annotations for the message payload in order to make it more robust.
+
+Example code of a system of systems with messaging can be found [here](https://github.com/olivergierke/sos/tree/master/30-messaging-sos){:target="_blank"}
+
+#### REST Messaging
+
+If you use REST you will have to deal with caching, pagination and conditional requests.
+
+Messages do not tend to be stored for long periods of time and most communication tends to be synchronous.
+One does have to pay attention on not to lose events as your application will immediately know if the message was processed correctly, incorrectly or timed out.
+
+#### REST Polling
+
+When using a polling mechanism, your producers do not send out messages to your consumers, but the consumers will poll the producers for new events to process.
+
+This means that:
+- You do not need any additional infrastructure like an Apache Kafka or a Message Bus
+- Event publication is part of the local transaction
+- The publishing system, producer, controls the lifecycle of the events and can transform these if necessary
+- The events never leave the publishing system
+- There might be a bigger consistency gap, depending on how frequently the consumers poll
+- It does not scale that well
+
+The example code with REST can be found [here](https://github.com/olivergierke/sos/tree/master/40-restful-sos){:target="_blank"}
+
+### Conclusion
+
+This was a great workshop which makes you think about the design decisions you have made for your applications.
+If you ever get the opportunity to participate in one of these workshops, do not hesitate to join as they are much more valuable than regular talks that can be viewed online as well.
+
+## Observability with Spring based distributed systems by [Tommy Ludwig](https://twitter.com/TommyLudwig){:target="_blank"}
+
+<span class="image left"><img class="p-image" alt="Tommy Ludwig" src="/img/spring-io-2018/tommy-ludwig.jpg"></span>
+
+### Introduction
+
+[Tommy](https://twitter.com/tommyludwig){:target="_blank"}'s talk introduced three main pillars of observability: logging, metrics, and tracing.
+
+Tommy explained that observability is achieved through a set of tools and practices that aim to turn data points and contexts into insights.
+Observability is something you should care about as it provides a great experience for the users of your system and it builds confidence in production where failure **will** happen.
+You ought to give yourself the tools you need in order to be a good owner in order to detect this failures as early as possible.
```

Comment:
-> "You ought to give yourself the tools you need in order to be a good owner so you can detect failures as early as possible."